File: Typescript/Encapsulation/without_encap.ts
class BankAccount {
    // Public variable
    balance: number = 0;

    deposit(amount: number): void {
        if (amount > 0) {
            this.balance += amount;
        }
    }

    withdraw(amount: number): void {
        if (amount > 0 && amount <= this.balance) {
            this.balance -= amount;
        }
    }
}

const acc = new BankAccount();

acc.deposit(5000);
acc.withdraw(1000);

//  ATM RULES BYPASSED
acc.balance = -9;

console.log("Balance:", acc.balance);

File: java/Encapsulation/withoutencap.java
class BankAccount {

    // ❌ PUBLIC DATA (BIG PROBLEM)
    public double balance;

    // Method exists
    public void deposit(double amount) {
        if (amount > 0) {
            balance += amount;
        }
    }

    public void withdraw(double amount) {
        if (amount > 0 && amount <= balance) {
            balance -= amount;
        }
    }
}

public class withoutencap {
    public static void main(String[] args) {
        BankAccount acc = new BankAccount();

        // Using ATM methods (GOOD)
        acc.deposit(5000);
        acc.withdraw(1000);

        // ❌ BYPASS ATM RULES
        acc.balance = -99999;   // illegal but allowed

        System.out.println("Balance: " + acc.balance);
    }
}
